src/dataset-builder/ConceptSearch.ts

```diff
@@ -105,39 +115,45 @@ export const ConceptSearch = (props: ConceptSearchProps) => {
-                name: div({ style: { display: 'flex' } }, [
-                  h(Link, { 'aria-label': `${label} ${concept.id}`, onClick: () => setCart(_.xor(cart, [concept])) }, [
-                    icon(iconName, { size: 16 }),
+            rows: _.map(
+              (concept) => {
+                const [label, iconName]: [string, IconId] = _.contains(concept, cart)
+                  ? ['remove', 'minus-circle-red']
+                  : ['add', 'plus-circle-filled'];
+                return {
+                  name: div({ style: { display: 'flex' } }, [
+                    h(
+                      Link,
+                      { 'aria-label': `${label} ${concept.id}`, onClick: () => setCart(_.xor(cart, [concept])) },
+                      [icon(iconName, { size: 16 })]
+                    ),
+                    div({ style: { marginLeft: 5 } }, [
+                      h(HighlightConceptName, { conceptName: concept.name, searchFilter: searchText }),
+                    ]),
                   ]),
-                  div({ style: { marginLeft: 5 } }, [
-                    h(HighlightConceptName, { conceptName: concept.name, searchFilter: search }),
+                  id: concept.id,
+                  count: concept.count,
+                  hierarchy: div({ style: { display: 'flex' } }, [
+                    h(
+                      Link,
+                      {
+                        'aria-label': `open hierarchy ${concept.id}`,
+                        onClick: () =>
+                          onOpenHierarchy(
+                            { id: concept.id, category: domainOption.category, root: concept },
+                            cart,
+                            searchText
+                          ),
+                      },
+                      [icon('view-list')]
+                    ),
+                    div({ style: { marginLeft: 5 } }, ['Hierarchy']),
                   ]),
-                ]),
-                id: concept.id,
-                count: concept.count,
-                hierarchy: div({ style: { display: 'flex' } }, [
-                  h(
-                    Link,
-                    {
-                      'aria-label': `open hierarchy ${concept.id}`,
-                      onClick: () =>
-                        onOpenHierarchy(
-                          { id: concept.id, category: domainOption.category, root: concept },
-                          cart,
-                          search
-                        ),
-                    },
-                    [icon('view-list')]
-                  ),
-                  div({ style: { marginLeft: 5 } }, ['Hierarchy']),
-                ]),
-              };
-            }, concepts.state.result),
+                };
+              },
+              // a.count is optional, we use `??` to handle undefined values, if undefined, replace with 0
+              concepts.state.result.sort((a, b) => (a.count ?? 0) - (b.count ?? 0))
```

Comment:
Can you explain this change to me? this seems like the only change here, and the rest is formatting changes.

Reply:
Im not sure, but I thought the sorting would happen in the back end once count is added.

Reply:
I added this because of the original ticket expected the concepts in descending order on the UI. I thought this meant the work needed to be done in the frontend. All the results have a count of 1?